packages/block-editor/src/components/block-toolbar/index.js

```diff
@@ -97,6 +98,7 @@ export function PrivateBlockToolbar( {
 
 			shouldShowVisualToolbar: isValid && isVisual,
 			rootClientId: blockRootClientId,
+			toolbarKey: `${ selectedBlockClientId }${ firstParentClientId }`,
```

Comment:
Why add a new selected key here? Can't we concatenate on render?

Reply:
We need access to the `firstParentClientId` outside of the selector, so we'll need to add a selected key of either the `firstParentClientId` or the concatenated `toolbarKey`. I can see how `firstParentClientId` would be more potentially useful than a `toolbarKey`.